src/mbgl/actor/scheduler.cpp

```diff
@@ -41,4 +41,30 @@ std::shared_ptr<Scheduler> Scheduler::GetBackground() {
     return scheduler;
 }
 
+// static
+std::shared_ptr<Scheduler> Scheduler::GetSequenced() {
+    const std::size_t kSchedulersCount = 10;
+    static std::vector<std::weak_ptr<Scheduler>> weaks(kSchedulersCount);
+    static std::mutex mtx;
+    static std::size_t lastUsedIndex = 0u;
+
+    if (++lastUsedIndex == kSchedulersCount) lastUsedIndex = 0u;
```

Comment:
This is not thread safe, should be after the lock.

Reply:
thanks for the catch!